ValveKeyValue/ValveKeyValue/ObjectCopier.cs

```diff
@@ -116,7 +116,7 @@ static KVValue ConvertObjectToValue(Type objectType, object managedObject, IObje
             }
             else
             {
-                foreach (var member in reflector.GetMembers(managedObject).OrderBy(p => p.Name, StringComparer.InvariantCulture))
+                foreach (var member in reflector.GetMembers(managedObject))
```

Comment:
Ordering in KV doesn't seem to affect anything, its largely a map/dictionary like JSON.

Reflection doesn't guarantee an order either, and even though members tend to be returned in declaration order the .NET docs state for GetMembers, GetProperties, and GetFields:

> The GetXXX method does not return members in a particular order, such as alphabetical or declaration order. Your code must not depend on the order in which members are returned, because that order varies.

For that reason I decided alphabetical was probably best here.

Reply:
If we strictly require ordering, potentially we could add an attribute for that, e.g. `[KVMember(Order = 1)]`?

Reply:
It also affects simple objects, e.g. here: https://github.com/SteamDatabase/ValveKeyValue/blob/master/ValveKeyValue/ValveKeyValue.Test/Text/SerializationTestCase.cs

not sure why that uses reflection in the first place

Reply:
Seems like the order will be preserved in NET7 by design? https://github.com/dotnet/runtime/pull/69506

I'd say removing OrderBy here is *fine*. Not sure if we need an order attribute, do other serializers do this?

Reply:
That PR appears to preserve the order returned by the native part of the CLR in the managed part of the CLR, but there is no strict guarantee that the native part of the CLR will return members in the order that you want.

As precedent, Newtonsoft.Json has JsonPropertyAttribute.Order and YamlDotNet has YamlMemberAttribute.Order.